File: frontend/src/types/project.ts
export interface Project {
  id: string
  name: string
  description?: string
  dialect?: string
  table_count: number
  relation_count: number
  created_at: string
  updated_at: string
}

export interface ProjectCreate {
  name: string
  description?: string
  dialect?: string
}

export interface UploadRequest {
  sql_content: string
}

export interface UploadResponse {
  tables_count: number
  relations_count: number
  errors: { statement_index: number; line: number; message: string }[]
}

export interface ProjectListResponse {
  items: Project[]
  total: number
  page: number
  size: number
}

File: backend/app/schemas/project.py
from datetime import datetime

from pydantic import BaseModel, Field


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None
    dialect: str = ""


class ProjectResponse(BaseModel):
    id: str
    name: str
    description: str | None = None
    dialect: str
    table_count: int = 0
    relation_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProjectListResponse(BaseModel):
    items: list[ProjectResponse]
    total: int
    page: int
    size: int


class UploadRequest(BaseModel):
    sql_content: str = Field(..., min_length=1)


class ErrorItem(BaseModel):
    statement_index: int
    line: int
    message: str


class UploadResponse(BaseModel):
    tables_count: int
    relations_count: int
    errors: list[ErrorItem] = []
